scripts/lib/javascriptifier.js

```diff
@@ -0,0 +1,164 @@
+  });
+
+  const idls = convertThriftFilesToJavascriptObject(allFileSpecs);
+  writeObjectAsJavascriptModule(idls, target);
+}
+
+/**
+ * Collect specifications for each file in a hierarchy starting from dirPath,
+ * relative to rootPath.
+ */
+function getFileSpecsRecursively(dirPath /*: string */, rootPath /*: string */) /*: FileSpec[] */ {
+  const files = readdirSync(dirPath);
+  const outFiles /*: string[] */ = [];
+  const children /*: string[] */ = [];
+
+  const fileData = files
+    .filter(fileName => {
+      const stat = lstatSync(`${dirPath}/${fileName}`);
+      if (stat.isDirectory()) {
+        children.push(fileName);
+        return false;
+      }
+      outFiles.push(fileName);
+      return true;
+    })
+    .map(fileName => {
+      const filePath = `${dirPath}/${fileName}`;
+      const relativePath = slash(path.relative(rootPath, filePath));
+      return {
+        fileName,
+        filePath,
+        relativePath,
+      };
+    });
+
+  // recursively call for all directories we identified before and append to the output
+  const childData = children.map(fileName => getFileSpecsRecursively(`${dirPath}/${fileName}`, rootPath));
+  childData.forEach(data => fileData.push(...data));
+  return fileData;
+}
+
+/**
+ * Given a set of file specs as returned by getFileSpecsRecursively, generate a jso
+ * with exports named for the relative path to each file, and returning the data
+ * in each file.
+ */
+function convertThriftFilesToJavascriptObject(fileSpecs /*: FileSpec[] */) {
+  const out = {};
+  fileSpecs.forEach(spec => {
+    const data = readFileSync(spec.filePath, 'utf-8');
+    out[spec.relativePath] = data;
+  });
+  return out;
+}
+
+/**
+ * Given a jso, write it as a javascript module file with exports named for each key.
+ * This expects the content to be thrift IDLs and strips comments from the data.
+ */
+function writeObjectAsJavascriptModule(obj /*: string */, targetPath /*: string */) {
+  const handle = openSync(targetPath, 'w');
+  writeSync(handle, getHeader());
+  Object.entries(obj).forEach(([key, value]) => {
+    const js = generateJavascriptKeyValuePair(key, value);
+    writeSync(handle, js);
+  });
+  writeSync(handle, getFooter());
+}
+
+/**
+ * Format a key/value pair as a javascript object descriptor
+ */
+function generateJavascriptKeyValuePair(name /*: string*/, data /*: string*/) {
+  const dataJsLiteral = '`' + escapeTemplateLiteral(stripCommentsFromThrift(data)) + '`,\n';
+  return `"${name}": ${dataJsLiteral}`;
+}
+
+function getHeader() {
+  return `
+/* eslint-disable camelcase */
+
+// GENERATED CODE - DO NOT MODIFY!
+
+module.exports = {
+`;
+}
+
+function getFooter() {
+  return '};\n';
+}
+
+// for windows
+function slash(somePath) {
+  return somePath.replace(/\\/g, '/');
+}
+
+const commentRegexp = /^\s*(#|\/\*|\*|\*\/|\/\/).*$/;
+const inlineCommentRegexp = /^([^"']+)\s(#|\/\/).*$/;
+function stripCommentsFromThrift(text) {
```

Comment:
What is the objective of stripping comments? To make the embedded string literal smaller?

Reply:
I mention this is the initial comment - yeah, it's an optimization. This change has the impact of making the IDLs become part of the memory footprint of this client, whereas they would not be when loaded & parsed dynamically. So the comments matter, and they are actually a fairly big part of the size of these files. 

It's probably not a big deal in the grand scheme - but seemed easy to just get rid of them. That said, there's a potential risk since this is a trivial way to remove them and doesn't involve actually parsing the syntax. I believe it to be safe, but if the spec for thrift IDL changes ever and there were some syntax (such as a non-comment leading slash or hash, possibly part of a multi-line statement?) then it might break. A better way seems like it would be to pre-parse the IDLs with `thriftrw` and emit some representation of the compiled AST instead of doing all this at all, but I didn't see any obvious way to "get compiled in and out" out of the thriftrw-node module, so that seemed to get way beyond the straightforward way of solving the basic problem that this PR seeks to solve.

In any event - stripping comments seems like a small risk, and I think if if failed, then tests would break, but not sure 🤷‍♂ ? Happy to do whatever you think is best here.

Reply:
ok